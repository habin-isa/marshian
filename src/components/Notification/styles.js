import styled, { keyframes } from 'styled-components';

const fadeIn = keyframes`
  from {
    opacity: 0;
    bottom: 0;
  }

  to {
    opacity: 1;
    bottom: 10px;
  }
`;

export const Wrapper = styled.div`
  margin: 16px;
  display: flex;
  justify-content: space-between;
  align-items: center;
  width: 500px;
  z-index: 9999;
  padding: 1em;
  background: #ffffff;
  box-shadow: 0px 1px 8px rgba(0, 0, 0, 0.05), 0px 0px 2px rgba(0, 0, 0, 0.3), 0px 3px 6px rgba(0, 0, 0, 0.2);
  border-radius: 4px;
  position: absolute;
  visibility: visible;
  animation: ${fadeIn} 800ms ease-in;
  transition: visibility 800ms ease-in;
  bottom: 10px;
  @media screen and (max-width: 430px) {
    width: 90%;
    height: auto;
  }
`;

export const TextContainer = styled.div`
  display: flex;
  flex-direction: column;
`;

export const Title = styled.span`
  font-weight: bold;
  font-size: 15.75px;
  line-height: 21px;
  color: #1e2337;
  white-space: nowrap;
  overflow: hidden;
  @media screen and (max-width: 430px) {
    font-size: 12.25px;
    line-height: 17px;
  }
`;

export const Subtitle = styled.span`
  font-size: 12.25px;
  line-height: 17px;
  color: #636674;
  @media screen and (max-width: 430px) {
    font-size: 12px;
    line-height: 10px;
  }
`;

export const IconContainer = styled.div`
  display: flex;
`;

export const CrossIcon = styled.img`
  padding: 10.5px;
  &:hover {
    opacity: 0.8;
  }
  cursor: pointer;
`;

export const ArrowIcon = styled.img`
  width: 40px;
  margin: 10.5px;
  padding-right: 10px;
  cursor: pointer;
  &:hover {
    opacity: 0.8;
  }
`;
