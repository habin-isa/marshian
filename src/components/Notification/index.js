import React from 'react';
import * as S from './styles';
import CrossIcon from '../../assets/close.svg';
import { func } from 'prop-types';

const Notification = ({ removeNotification }) => {
  return (
    <S.Wrapper>
      <S.IconContainer></S.IconContainer>
      <S.TextContainer>
        <S.Title>Hey there!</S.Title>
        <S.Subtitle>
          Welcome to Marshian, we hit the SpaceX API to cater your spacecraft needs. Please filter between 'Dragons' or
          'Rockets', clicking on the card will show more details.
        </S.Subtitle>
      </S.TextContainer>
      <S.IconContainer>
        <S.CrossIcon src={CrossIcon} alt="cross-icon" onClick={() => removeNotification()} />
      </S.IconContainer>
    </S.Wrapper>
  );
};

export default Notification;

Notification.propTypes = {
  removeNotification: func
};

Notification.defaultProps = {
  removeNotification: () => {}
};
