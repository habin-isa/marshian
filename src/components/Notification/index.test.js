import React from 'react';
import { shallow } from 'enzyme';
import { render } from '@testing-library/react';
import Notification from '.';

describe('Notification component', () => {
  const props = {
    removeNotification: () => {}
  };
  it('renders without crashing', () => {
    const { container } = render(<Notification removeNotification={props.removeNotification} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders with correct user data', () => {
    shallow(<Notification removeNotification={props.removeNotification} />);
  });
});
