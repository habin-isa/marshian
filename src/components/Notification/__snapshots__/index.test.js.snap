// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification component renders without crashing 1`] = `
<div
  class="sc-bdnylx gvncCt"
>
  <div
    class="sc-eCApGN hCdAiT"
  />
  <div
    class="sc-gtssRu bMBAbX"
  >
    <span
      class="sc-dlnjPT bLahkL"
    >
      Hey there!
    </span>
    <span
      class="sc-hKFyIo DMNNq"
    >
      Welcome to Marshian, we hit the SpaceX API to cater your spacecraft needs. Please filter between 'Dragons' or 'Rockets', clicking on the card will show more details.
    </span>
  </div>
  <div
    class="sc-eCApGN hCdAiT"
  >
    <img
      alt="cross-icon"
      class="sc-jSFkmK gdTyUL"
      src="close.svg"
    />
  </div>
</div>
`;
